homeassistant/components/comelit/__init__.py

```diff
@@ -1,40 +1,87 @@
 """Comelit integration."""
 
+from typing import Any
+
+from aiocomelit.const import BRIDGE
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_HOST, CONF_PIN, CONF_PORT, Platform
+from homeassistant.const import CONF_HOST, CONF_PIN, CONF_PORT, CONF_TYPE, Platform
 from homeassistant.core import HomeAssistant
 
 from .const import DEFAULT_PORT, DOMAIN
-from .coordinator import ComelitSerialBridge
+from .coordinator import ComelitSerialBridge, ComelitVedoSystem
 
-PLATFORMS = [Platform.COVER, Platform.LIGHT, Platform.SENSOR, Platform.SWITCH]
+BRIDGE_PLATFORMS = [
+    Platform.COVER,
+    Platform.LIGHT,
+    Platform.SENSOR,
+    Platform.SWITCH,
+]
+VEDO_PLATFORMS = [
+    Platform.ALARM_CONTROL_PANEL,
+]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Comelit platform."""
-    coordinator = ComelitSerialBridge(
-        hass,
-        entry.data[CONF_HOST],
-        entry.data.get(CONF_PORT, DEFAULT_PORT),
-        entry.data[CONF_PIN],
-    )
+
+    coordinator: Any
+    if entry.data.get(CONF_TYPE, BRIDGE) == BRIDGE:
+        coordinator = ComelitSerialBridge(
+            hass,
+            entry.data[CONF_HOST],
+            entry.data.get(CONF_PORT, DEFAULT_PORT),
+            entry.data[CONF_PIN],
+        )
+        platforms = BRIDGE_PLATFORMS
+    else:
+        coordinator = ComelitVedoSystem(
+            hass,
+            entry.data[CONF_HOST],
+            entry.data.get(CONF_PORT, DEFAULT_PORT),
+            entry.data[CONF_PIN],
+        )
+        platforms = VEDO_PLATFORMS
 
     await coordinator.async_config_entry_first_refresh()
 
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
 
-    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    await hass.config_entries.async_forward_entry_setups(entry, platforms)
 
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
Since the superclass `ComelitCommonApi` contains both `.logout()` and `.close()` you can make this one function again and then only check what type the entry is to define which platforms to unload

Reply:
Done